src/__tests__/__views__/SignUp.test.js

```diff
@@ -0,0 +1,151 @@
+
+const setup = () => {
+  const props = {
+    handleChange: jest.fn(),
+    handleSubmit: jest.fn(),
+    handleBlur: jest.fn(),
+    user: {
+      email: "test",
+      username: "test",
+      password: "test",
+      confirmPassword: "test"
+    },
+    submitted: false,
+    message: "",
+    success: false,
+    dispatch: jest.fn(),
+    history: []
+  };
+
+  return {
+    props
+  };
+};
+describe("Render the sign up page", () => {
+  const { props } = setup();
+
+  const registerComponent = mount(
+    <MemoryRouter>
+      <SignUp {...props} />
+    </MemoryRouter>
+  ).find("SignUp");
+  it("snapshot testing for the component", () => {
+    expect(toJson(registerComponent)).toMatchSnapshot();
+    expect(registerComponent).toBeTruthy();
+  });
+
+  it("should render 3 text input and a button", () => {
+    expect(registerComponent.find("input").length).toBe(4);
+    expect(
+      registerComponent
+        .find("input")
+        .at(3)
+        .props().value
+    ).toEqual("Sign Up");
+  });
+  it("should not display error message when rendering the  component", () => {
+    expect(registerComponent.find(".auth-errors").length).toBeFalsy();
+  });
+
+  it("should display error message when there is an error ", () => {});
+
+  describe("test event", () => {
+    let instance;
+    beforeEach(() => {
+      instance = registerComponent.find("SignUp").instance();
+      jest.spyOn(instance, "handleChange");
+      jest.spyOn(instance, "handleSubmit");
+      jest.spyOn(instance, "handleBlur");
+      instance.forceUpdate();
+    });
+    afterEach(() => {
+      jest.restoreAllMocks();
+    });
+    it("should handle on change for username", () => {
+      const usernameInput = registerComponent.find("input").at(0);
+      usernameInput.simulate("change", {
+        target: {
+          target: "username",
+          value: "password"
+        }
+      });
+      expect(instance.handleChange).toHaveBeenCalled();
+    });
+
+    it("should handle on change for email", () => {
+      const usernameInput = registerComponent.find("input").at(0);
+      usernameInput.simulate("change", {
+        target: {
+          target: "email",
+          value: "esp@email.com"
+        }
+      });
+      expect(instance.handleChange).toHaveBeenCalled();
+    });
+
+    it("should handle on change for password", () => {
+      const usernameInput = registerComponent.find("input").at(0);
+      usernameInput.simulate("change", {
+        target: {
+          target: "password",
+          value: "password"
+        }
+      });
+      expect(instance.handleChange).toHaveBeenCalled();
+    });
+
+    it("should handle on  blur on all field", () => {
+      registerComponent.find("input").forEach(inputField => {
+        inputField.simulate("blur", {
+          target: {
+            target: "test",
+            value: "test"
+          }
+        });
+        expect(instance.handleBlur).toHaveBeenCalled();
+      });
+    });
```

Comment:
An empty line is required after line 113 before starting a new case scenario for better readability.